src/types/decimal.rs

```diff
@@ -296,8 +296,42 @@ impl TryFrom<f64> for Decimal {
 
 impl Display for Decimal {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        // TODO: This is correct, but not the most human-friendly format.
-        write!(f, "{}d{}", self.coefficient, self.exponent)
+        let digits = &*self.coefficient.magnitude.to_string();
+        let len = digits.len();
+        // The index of the decimal, relative to the magnitude representation
```

Comment:
```suggestion
        // The index of the decimal point, relative to the magnitude representation
```
Given that we're working with a `Decimal` data type here, I think we should be more explicit that this refers to the decimal _point_. (It does, right?)